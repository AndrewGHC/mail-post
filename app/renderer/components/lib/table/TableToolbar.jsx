import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import DeleteIcon from '@material-ui/icons/Delete';
import { lighten } from '@material-ui/core/styles/colorManipulator';

const toolbarStyles = theme => ({
  root: {
    paddingRight: theme.spacing.unit,
  },
  highlight: {
    color: theme.palette.secondary.main,
    backgroundColor: lighten(theme.palette.secondary.light, 0.85),
  },
  spacer: {
    flex: '1 1 100%',
  },
  actions: {
    color: theme.palette.text.secondary,
  },
  title: {
    flex: '0 0 auto',
  },
});

const EnhancedTableToolbar = (props) => {
  const {
    numSelected,
    classes,
    onClickDelete,
    title,
  } = props;

  return (
    <Toolbar
      className={classNames(classes.root, {
        [classes.highlight]: numSelected > 0,
      })}
    >
      <div className={classes.title}>
        {numSelected > 0 ? (
          <Typography color="inherit" variant="subtitle1">
            {numSelected}
            {' '}
selected
          </Typography>
        ) : (
          <Typography variant="h6">{title}</Typography>
        )}
      </div>
      <div className={classes.spacer} />
      <div className={classes.actions}>
        {numSelected > 0 && (
          <Tooltip title="Delete">
            <IconButton aria-label="Delete" onClick={onClickDelete}>
              <DeleteIcon />
            </IconButton>
          </Tooltip>
        )}
      </div>
    </Toolbar>
  );
};

EnhancedTableToolbar.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  numSelected: PropTypes.number.isRequired,
  onClickDelete: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
};

const StyledEnhancedTableToolbar = withStyles(toolbarStyles)(EnhancedTableToolbar);

export default StyledEnhancedTableToolbar;
