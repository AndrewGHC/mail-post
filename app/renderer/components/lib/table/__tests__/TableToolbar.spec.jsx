import React from 'react';
import td from 'testdouble';
import { mount } from 'enzyme';
import IconButton from 'material-ui/IconButton';

import TableToolbar from '../TableToolbar';

require('testdouble-jest')(td, jest);

describe('TableToolbar', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      classes: {},
      numSelected: 0,
      onClickDelete: td.function(),
      title: 'Test Table',
    };
    wrapper = mount(<TableToolbar {...props} />);
  });

  it('matches snapshot', async () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('when numSelected is NOT > 0, renders title', async () => {
    expect(wrapper.text().includes(props.title)).toBe(true);
  });

  it('when numSelected > 0, renders "{n} selected" heading', async () => {
    const numSelected = 8;
    wrapper = mount(<TableToolbar {...props} numSelected={numSelected} />);
    expect(wrapper.text().includes(`${numSelected} selected`)).toBe(true);
  });

  it('when numSelected > 0 AND IconButton is clicked, calls prop onClickDelete', async () => {
    const numSelected = 8;
    wrapper = mount(<TableToolbar {...props} numSelected={numSelected} />);
    wrapper.find(IconButton).simulate('click');
    td.verify(props.onClickDelete(td.matchers.anything()));
  });
});
