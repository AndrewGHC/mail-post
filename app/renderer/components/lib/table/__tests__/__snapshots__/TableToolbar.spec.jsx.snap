// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableToolbar matches snapshot 1`] = `
<WithStyles(EnhancedTableToolbar)
  classes={Object {}}
  numSelected={0}
  onClickDelete={[Function]}
  title="Test Table"
>
  <EnhancedTableToolbar
    classes={
      Object {
        "actions": "EnhancedTableToolbar-actions-4",
        "highlight": "EnhancedTableToolbar-highlight-2",
        "root": "EnhancedTableToolbar-root-1",
        "spacer": "EnhancedTableToolbar-spacer-3",
        "title": "EnhancedTableToolbar-title-5",
      }
    }
    numSelected={0}
    onClickDelete={[Function]}
    title="Test Table"
  >
    <WithStyles(Toolbar)
      className="EnhancedTableToolbar-root-1"
    >
      <Toolbar
        className="EnhancedTableToolbar-root-1"
        classes={
          Object {
            "gutters": "MuiToolbar-gutters-7",
            "root": "MuiToolbar-root-6",
          }
        }
        disableGutters={false}
      >
        <div
          className="MuiToolbar-root-6 MuiToolbar-gutters-7 EnhancedTableToolbar-root-1"
        >
          <div
            className="EnhancedTableToolbar-title-5"
          >
            <WithStyles(Typography)
              variant="title"
            >
              <Typography
                align="inherit"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter-21",
                    "alignJustify": "MuiTypography-alignJustify-23",
                    "alignLeft": "MuiTypography-alignLeft-20",
                    "alignRight": "MuiTypography-alignRight-22",
                    "body1": "MuiTypography-body1-17",
                    "body2": "MuiTypography-body2-16",
                    "button": "MuiTypography-button-19",
                    "caption": "MuiTypography-caption-18",
                    "colorError": "MuiTypography-colorError-31",
                    "colorInherit": "MuiTypography-colorInherit-27",
                    "colorPrimary": "MuiTypography-colorPrimary-28",
                    "colorSecondary": "MuiTypography-colorSecondary-29",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary-30",
                    "display1": "MuiTypography-display1-12",
                    "display2": "MuiTypography-display2-11",
                    "display3": "MuiTypography-display3-10",
                    "display4": "MuiTypography-display4-9",
                    "gutterBottom": "MuiTypography-gutterBottom-25",
                    "headline": "MuiTypography-headline-13",
                    "noWrap": "MuiTypography-noWrap-24",
                    "paragraph": "MuiTypography-paragraph-26",
                    "root": "MuiTypography-root-8",
                    "subheading": "MuiTypography-subheading-15",
                    "title": "MuiTypography-title-14",
                  }
                }
                color="default"
                gutterBottom={false}
                headlineMapping={
                  Object {
                    "body1": "p",
                    "body2": "aside",
                    "display1": "h1",
                    "display2": "h1",
                    "display3": "h1",
                    "display4": "h1",
                    "headline": "h1",
                    "subheading": "h3",
                    "title": "h2",
                  }
                }
                noWrap={false}
                paragraph={false}
                variant="title"
              >
                <h2
                  className="MuiTypography-root-8 MuiTypography-title-14"
                >
                  Test Table
                </h2>
              </Typography>
            </WithStyles(Typography)>
          </div>
          <div
            className="EnhancedTableToolbar-spacer-3"
          />
          <div
            className="EnhancedTableToolbar-actions-4"
          />
        </div>
      </Toolbar>
    </WithStyles(Toolbar)>
  </EnhancedTableToolbar>
</WithStyles(EnhancedTableToolbar)>
`;
